### PR TITLE
Modifications to build on windows with msvc.

### PR DESCRIPTION
**Pull request by tomlandrum**

### CODE REVIEW

File: src/interface/ui/accountactivity/BlockchainAccountActivity.cpp [hunk 1]
```diff
@@ -308,7 +308,7 @@ auto BlockchainAccountActivity::load_thread_portion_p() noexcept -> bool
 
     PTimer<Diag>::start(Diag::portion_p);
     tdiag("PPP 1");
-    std::vector<std::future<unsigned>> fvec = [this]() {
+    std::vector<std::future<unsigned>> fvec = [=]() {
```
Comment (wiesiekpap) [inline src/interface/ui/accountactivity/BlockchainAccountActivity.cpp:311]: Another way would be [this,threadcount,stride] - a bit more verbose but the compiler will avoid copying several of the data fields that the lambda needs.

**Review by wiesiekpap — APPROVED** [review]

**Comment by wiesiekpap** [comment]
@tomlandrum I have closed my PR #150 as it contained a subset of your changes here and it would have caused an unnecessary round of merge conflict resolution.

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,4 @@
 cmake-build-debug/*
 src/cmake-build-debug
 
+.vs
```
Comment (puszupek) [inline .gitignore:11]: Please, move it higher near .vscode (or even remove it and do it only locally in .git/info/exclude).

File: src/api/Legacy.cpp [hunk 3]
```diff
@@ -223,9 +223,9 @@ auto Legacy::AppendFolder(String& out, const String& base, const String& file)
     const noexcept -> bool
 {
     try {
-        const auto path = fs::path{base.Get()}.remove_trailing_separator() /
-                          fs::path{file.Get()}.remove_trailing_separator() /
-                          fs::path{"/"};
+        auto path = fs::path{ base.Get() }.remove_trailing_separator() /
+                    fs::path{ file.Get() }.remove_trailing_separator();
+        path += fs::path{"/"};
```
Comment (puszupek) [inline src/api/Legacy.cpp:228]: What was the problem here? Isn't the problem that we have `/` in `fs::path{"/"}` and for Windows shouldn't we have `\`?

File: src/blockchain/block/Position.cpp [hunk 4]
```diff
@@ -29,7 +29,7 @@ Position::Position(const Height& height, const Hash& hash) noexcept
 }
 
 Position::Position(const Height& height, Hash&& hash) noexcept
-    : Position(Height{height}, std::move(hash))
+    : Position(std::move(Height{height}), std::move(hash))
```
Comment (puszupek) [inline src/blockchain/block/Position.cpp]: I don't get this change. For me `std::move(Height{height})` is r-value reference and `Height{height}` is also r-value reference and should gave the same output and invoke the same constructor.

File: src/interface/ui/blockchainselection/BlockchainSelection.cpp [hunk 5]
```diff
@@ -113,8 +113,8 @@ BlockchainSelection::BlockchainSelection(
 
         return out;
     }())
-    , enabled_count_(0)
-    , enabled_callback_()
+    , enabled_count_{}
+    , enabled_callback_{}
     , last_job_{}
```
Comment (puszupek) [inline src/interface/ui/blockchainselection/BlockchainSelection.cpp]: There is another construktor above. You can change it also like here.

**Pull request merged**